pkg/local_object_storage/engine/engine.go

```diff
@@ -53,12 +54,25 @@ func (e *StorageEngine) reportShardError(
 
 	err = sh.SetMode(mode.DegradedReadOnly)
 	if err != nil {
-		e.log.Error("failed to move shard in degraded mode",
+		e.log.Error("failed to move shard in degraded-read-only mode, moving to read-only",
+			zap.Stringer("shard_id", sid),
 			zap.Uint32("error count", errCount),
 			zap.Error(err))
+
+		err = sh.SetMode(mode.ReadOnly)
```

Comment:
Can you please explain why setting `ReadOnly` can succeed after the `DegradedReadOnly` failed (not obvious at first glance)?

Also maybe it's worth to mention the fallback in `reportShardError` docs.